classes/core/PKPHtmlSanitizer.php

```diff
@@ -0,0 +1,192 @@
+<?php
+
+/**
+ * @file classes/core/PKPHtmlSanitizer.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2000-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class PKPHtmlSanitizer
+ *
+ * @ingroup core
+ *
+ * @brief Wrapper on the top of Symfony's HtmlSanitizer implementation
+ *
+ */
+
+namespace PKP\core;
+
+use Illuminate\Support\Str;
+use Illuminate\Support\Collection;
+use PKP\config\Config;
+use Symfony\Component\HtmlSanitizer\HtmlSanitizer;
+use Symfony\Component\HtmlSanitizer\HtmlSanitizerConfig;
+use Symfony\Component\HtmlSanitizer\Reference\W3CReference;
+
+class PKPHtmlSanitizer
+{
+    /**
+     * Collection of valid element specify by W3C Sanitizer API
+     * 
+     * @see const HEAD_ELEMENTS and BODY_ELEMENTS at \Symfony\Component\HtmlSanitizer\Reference\W3CReference
+     * @see https://wicg.github.io/sanitizer-api/#default-configuration
+     */
+    protected Collection $w3cValidElements;
+
+    /**
+     * Instance of HtmlSanitizerConfig config
+     */
+    protected HtmlSanitizerConfig $htmlSanitizerConfig;
+
+    /**
+     * Instance of HtmlSanitizer
+     */
+    protected HtmlSanitizer $htmlSanitizer;
+
+    /**
+     * Create a new instance
+     * 
+     * @param string $allowable Config key such security.[allowed_html/allowed_title_html]
+     *                          or string of allowed tags with attribited generated in same
+     *                          structure as the security.[allowed_html/allowed_title_html]
+     */
+    public function __construct(string $allowable)
+    {
+        $this->w3cValidElements = collect(
+            array_merge(W3CReference::HEAD_ELEMENTS, W3CReference::BODY_ELEMENTS)
+        )->keys();
+
+        $this->htmlSanitizer = new HtmlSanitizer(
+            $this->buildSanitizerConfig(
+                $this->generateAllowedTagToAttributeMap(
+                    Config::getVar('security', $allowable, null) ?? $allowable
+                )
+            )
+        );
+    }
+
+    /**
+     * Set a pre defined instance of \Symfony\Component\HtmlSanitizer\HtmlSanitizerConfig
+     */
+    public function setSanitizerConfig(HtmlSanitizerConfig $htmlSanitizerConfig): self
```

Comment:
Are the getter/setter ever used? It doesn't look like it (and the recreation of `htmlSanitizer` seems like an opportunity for side-effects).

Reply:
I am thinking if from more of a generalized class that can be reuse other places so make the `HtmlSanitizerConfig` modifiable but we can remove those as those not needed anywhere right now and later introduce it if ever needed to .